airflow-core/src/airflow/api_fastapi/core_api/security.py

```diff
@@ -62,16 +64,33 @@
     "information (such as user identity and scope) to authenticate subsequent requests. "
     "To learn more about Airflow public API authentication, please read https://airflow.apache.org/docs/apache-airflow/stable/security/api.html."
 )
-oauth2_scheme = OAuth2PasswordBearer(tokenUrl="/auth/token", description=auth_description)
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl="/auth/token", description=auth_description, auto_error=False)
```

Comment:
I changed option `auto_error=False` because resolve_user_from_token role is to raise error.